src/Server.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Server.hpp"
+#include "nlohmann/json.hpp"
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include "curl/curl.h"
+#include "Utility.hpp"
+#include "User.hpp"
+
+std::string data;
+
+size_t write_data(char* buf, size_t size, size_t nmemb, void* up)
+{
+    for (size_t c = 0; c < size*nmemb; c++)
+        data.push_back(buf[c]);
+    return size * nmemb;
+}
+
+void Server::downloadUsers()
+{
+    CURL* curl;
+    curl_global_init(CURL_GLOBAL_ALL);
+    curl = curl_easy_init();
+
+    if (curl)
+    {
+        std::string path = serverName + pathToFile;
+        curl_easy_setopt(curl, CURLOPT_URL, path.c_str());
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, &write_data);
+        curl_easy_perform(curl);
+
+        if (checkIfFileExist())
+        {
+            Users listOfUsers = convertToArray(data);
```

Comment:
It's a local variable that is not returned, so after downloading nothing happeds. Users should be put in the database.